### PR TITLE
[FIX] point_of_sale: sale details report

### PR DESCRIPTION
**Pull request by rhe-odoo**
Due to legal considerations, it is necessary to modify the template for certain fields.

We also fix the display of differents values because it was not corretcly displayed. Additionally, rectifications have been made to accurately display various values that were previously incorrect.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/views/report_saledetails.xml [hunk 1]
```diff
@@ -373,22 +363,46 @@
...
+                                    <span t-out='currency["symbol"]'>$</span><span t-out='currency["total_paid"]' t-options="{'widget': 'float', 'precision': currency['precision']}">99.99</span>
+                                </span></td>
+                            </tr>
+                            <tr>
+                                <td>Total with rounding</td>
+                                <td class="text-end"><span t-if="currency['position']">
+                                    <span t-out="total_paid" t-options="{'widget': 'float', 'precision': currency['precision']}">99.99</span><span t-out='currency["symbol"]'>$</span>
+                                </span>
+                                <span t-else="">
+                                    <span t-out='total_paid'>$</span><span t-out='currency["total_paid"]' t-options="{'widget': 'float', 'precision': currency['precision']}">99.99</span>
```
Comment (adgu-odoo) [inline addons/point_of_sale/views/report_saledetails.xml:387]:
```suggestion
                                    <span t-out='currency["symbol"]'>$</span><span t-out='total_paid' t-options="{'widget': 'float', 'precision': currency['precision']}">99.99</span>
```

File: addons/point_of_sale/views/report_saledetails.xml [hunk 2]
```diff
@@ -373,22 +363,46 @@
...
+                                </span></td>
+                            </tr>
+                            <tr>
+                                <td>Rounding difference</td>
+                                <td class="text-end">
+                                <span t-if="currency['position']">
+                                    <span t-out="currency['total_paid'] - total_paid"  t-options="{'widget': 'float', 'precision': currency['precision']}">99.99</span><span t-out='currency["symbol"]'>$</span>
+                                </span>
+                                <span t-else="">
+                                    <span t-out="currency['total_paid'] - total_paid" >$</span><span t-out='currency["total_paid"]' t-options="{'widget': 'float', 'precision': currency['precision']}">99.99</span>
```
Comment (adgu-odoo) [inline addons/point_of_sale/views/report_saledetails.xml:397]:
```suggestion
                                    <span t-out="currency["symbol"]" >$</span><span t-out='currency["total_paid"] - total_paid' t-options="{'widget': 'float', 'precision': currency['precision']}">99.99</span>
```

**Review by adgu-odoo — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js [hunk 3]
```diff
@@ -276,6 +276,7 @@ export class PaymentScreen extends Component {
             // 1. Save order to server.
             syncOrderResult = await this.pos.push_single_order(this.currentOrder);
             if (!syncOrderResult) {
+                this.currentOrder.finalized = false;
```
Comment (caburj) [inline addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js:279]: move this in the override of `push_single_order` after calling `pushToBlackbox`.

File: addons/test_mail/tests/test_mail_template.py [hunk 4]
```diff
@@ -242,7 +242,7 @@ def test_template_send_email_wreport(self):
         """ Test 'send_email' on template on a given record, used notably as
         contextual action, with dynamic reports involved """
         self.env.invalidate_all()
-        with self.with_user(self.user_employee.login), self.assertQueryCount(107):  # test_mail: 106
+        with self.with_user(self.user_employee.login), self.assertQueryCount(109):  # test_mail: 106
```
Comment (tde-banana-odoo) [inline addons/test_mail/tests/test_mail_template.py:245]: :thinking:

**Review by caburj — APPROVED** [review]
robodoo r+

**Pull request closed without merging**